### ISSUE TITLE
[Bug] ROI of FoR_1 temporarily shows in FoR_2  

### ISSUE THREAD

**Issue by jenny-hm-lee** [OHIF/Viewers#3909]
### Describe the Bug

**Affected Version**:  OHIF version 3.8.0-beta.51

In a study, say we have the following series
* CT from frame of reference 1, (FoR1)
* RTSTRUCT created on FoR 1
* CT from frame of reference 2, (FoR2)
* RTSTRTUCT created on FoR2

When loading the above CT/RTSTRUCT pairs in two viewports, the ROI sometimes display in the incorrect viewport.

### Steps to Reproduce

See below video, and listen to my audio

https://app.screencastify.com/v3/watch/Zmb1P4seLoskQWhGatzg

### The current behavior

See below video, and listen to my audio

### The expected behavior

When user clicks on an ROI on the segmentation panel, the associated viewport should be highlighted, and ROI should be displayed correctly in each of their frame of reference.  

### OS

Windows 10

### Node version

v16.17.0

### Browser

Chrome Version 114.0.5735.110

**Comment by sedghi** [comment]
This seems to be related to the click, and highlight potentially

**Comment by jenny-hm-lee** [comment]
yes, this is still an issue at 3.9.beta.2.  The buggy behavior is different.

### Steps:

1. load the sample study
2. set up 2 viewports
3. on your left viewport, load "CT Std (FoR1)"
4. on your right viewport, load "CT Std (FoR 2)"
5. on your left viewport, load "FoR 1 RTstruct".  
6. Expect FoR 1 RTstruct to show on left viewport.  Actually, it shows on your right viewport.





### Below is the sample dataset:
https://fileshare.uhn.ca/download/e8daa740-b547-413d-a668-c3a825d9b32c
Please note that this file upload will expire in 30 days.
Password is sentence

**Comment by BrandonDatUHN** [comment]
Additional images for context
after step 5
![image](https://github.com/OHIF/Viewers/assets/112980389/a758e033-1789-4178-a444-8c586940726b)
after clicking Yes to open segmentation
![image](https://github.com/OHIF/Viewers/assets/112980389/f03ed7b7-efb4-4947-9ec1-e1c464787917)

**Comment by sedghi** [comment]
Is this still an issue

**Comment by sedghi** [comment]
@jenny-hm-lee Can you assist us in determining if this issue persists?

**Comment by BrandonDatUHN** [comment]
Still an issue on 3.9.0-beta 43 on OHIF dev local using the fileshare link above

It no longer loads all slices of the RTstruct on one slice but it is loading it on the wrong image and panel.

![image](https://github.com/OHIF/Viewers/assets/112980389/85e35b83-f207-4941-a0cd-9f9dee6edf6c)

**Comment by IbrahimCSAE** [comment]
@BrandonDatUHN can I get a new fileshare link? Its expired.

**Comment by jenny-hm-lee** [comment]
@IbrahimCSAE 
https://fileshare.uhn.ca/download/717ae753-a2c3-412c-8487-f7bde485b2f2

**Comment by sedghi** [comment]
We just released version 3.9 (still in beta) with Cornerstone3D 2.0. Could you try it out and check if this issue persists? Also, let us know if you encounter any regressions.

Viewer-dev.ohif.org
